types/hooks.d.ts

```diff
@@ -200,6 +200,18 @@ export interface onRegisterHookHandler<
   ): Promise<unknown> | void // documentation is missing the `done` method
 }
 
+/**
+ * Triggered when fastify.listen() or fastify.ready() is invoked to start the server. It is useful when plugins need a "ready" event, for example to load data before the server start listening for requests.
+ */
+export interface onReadyHookHandler<
```

Comment:
@Ethan-Arrowood I should have added the types.

I removed the `RawRequest` and `RawReply` from the generic interface since in that hook the user can't use those object.

I'm not sure, but also in the `onClose` can they be removed?

Reply:
`onClose` has the instance argument which we want to reflect the existing Fastify instance (so we need to pass through those generics). If that argument no longer exists then yes it can be removed as well.

Reply:
I would open an issue for that because I think it is out of scope for this PR and it doesn't look so simple to me to archive it because for what I understand:

- `onClose` receives the fastify instance as the first parameter
- the fastify instance in TS is defined as `FastifyInstance<RawServer, RawRequest, RawReply, Logger>`

So I can't just remove them without change here:

```
  (
    instance: FastifyInstance<RawServer, RawRequest, RawReply, Logger>, // need to define another `FastifyInstance`
    done: HookHandlerDoneFunction
  ): Promise<unknown> | void
```

I tried with `null` but get:

```
 ×  223:41  Type null does not satisfy the constraint RawRequestDefaultExpression<RawServer>.
```

So it seems to me that we should define a new `FastifyInstance` with fewer generics parameter and would it lead to many commits?

Reply:
All i'm wondering is if the `onClose` hook handler still has the `instance` argument. If it doesn't have that argument any longer than we can remove the types; otherwise, no change is needed afaik

Reply:
Yes, it has the instance args